src/Transform/ONNX/ConstProp.td

```diff
@@ -185,6 +187,25 @@ def SqrtofConst :  Pat<
 // Exactly the same pattern as for the elementwise ADD operations.
 //===----------------------------------------------------------------------===//
 
+def ReduceZeroAdd : Pat<
```

Comment:
Add test case where x is broadcasted to dense zero tensor's shape, followed by MatMul.
For example, something like:
```
%2 = onnx.constant()  {value = dense<0> : <320xf32>} : () -> tensor<320xf32>
%0 = onnx.add(%1, %2) : (tensor<1xf32>, tensor<320xf32>) -> tensor<320xf32>
%3 = TMatMul(%0, %4) : (tensor<320xf32>, tensor<320xf32>) -> tensor<1xf32>
```

Reply:
If above case fails, we might want to only do ReduceZeroAdd if x doesn't broadcast to zero tensor's shape.

Reply:
You are right that test fails, I'm just going to get rid of the add by 0 in quantize/dequantize lowering.

Reply:
Also added the test into the end_to_end tests in this mr : https://git.groq.io/code/Groq/-/merge_requests/9322